### ISSUE TITLE
problems with ADFW gateway

### ISSUE THREAD

**Issue by svilchezl** [robertsLando/node-red-contrib-m-bus#20]
Hi, 
I have an adfw gateway HD67030-B2-160, and I can't see more than one counter... with mbsheet I recognize all the devices, about 140, and they all have the primary address 0, but the ID that they have. 
How can I search for equipment by ID?
With the Scan order, it only finds 1 counter.

Thanks,

**Comment by robertsLando** [comment]
Search by secondary address

**Comment by svilchezl** [comment]
Yes, but I don't know how to do it with the example flow, can it be done?

**Comment by robertsLando** [comment]
Just send a msg with topic `scan` in input to controller

**Comment by svilchezl** [comment]
I have quite a few problems locating the counters, since I can't locate more than 70 when in fact there are about 140... they also get lost many times. I adjusted the values to the ones I had in the previous RELAY interface but there is no way to make the readings stable... I don't know where the problem can come from, if it is from the ADF communication with mbus, or towards nodered... there is any way to check?

**Comment by robertsLando** [comment]
I personally always had bad times with m-bus protocl. It could be there are address conflicts or there could be some other compatiblity issue on gateway too

**Comment by svilchezl** [comment]
Hir Roberts,

In principle there are no conflicts, this installation was already working with an Intesis mbus-modbus gateway, but we have updated the system to send the data to KNX, which is where the current BMS. 
The problem is that what I am seeing is not reliable and I would like to analyze the problem. 
In other installations we used its direct driver via RS232 and it worked, but here is the first experience with direct LAN... yes, we had much fewer units.
